### PR TITLE
Kivanov/custom dam provider

### PR DESCRIPTION
**Pull request by P1ndl3sk1n**

### CODE REVIEW

**Comment by DimitarGaydardzhiev** [comment]
I think this pull request should be to the "develop" branch

File: src/library-extender/README.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        ...
+    }
+    
+    loadMediaSelector(damWrapper: HTMLElement, mediaEntityData: EntityData, allowMultiSelect: boolean): void {
+        ...
+    }
+}    
+```
+There are two abstract methods that need to be implemented because they are specific for each provider:
+* isSupported - Use this method to determine if your provider supports the one that is enabled on the server. As parameter it receives the name of the provider which is defined server-side.
```
Comment (FerdiNebi) [inline src/library-extender/README.md:33]: "it receives the name of the provider which is defined" -> It receives the name of the Provider Type. You can name the provider anyway you want, what matters is the name of the Class on the server.

File: src/library-extender/README.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        }
+
+        let config = {
+            cloud_name: this.getPropertyValue("CloudName"),
+            api_key: this.getPropertyValue("ApiKey"),
+            multiple: allowMultiSelect,
+            inline_container: `.${damWrapper.className.replace(/\s/g, ".")}`,
+            remove_header: true,
+            integration: {
+                type: "custom_progress_sitefinity_connector_for_frontify",
```
Comment (FerdiNebi) [inline src/library-extender/README.md:61]: custom_progress_sitefinity_connector_for_frontify => for cloudinary?

File: src/library-extender/dam/custom-dam-provider.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+            // create configuration specific for Cloudinary
+            const config = {
+                cloud_name: this.getPropertyValue("CloudName"),
+                api_key: this.getPropertyValue("ApiKey"),
+                multiple: allowMultiSelect,
+                inline_container: `.${damWrapper.className.replace(/\s/g, ".")}`,
+                remove_header: true,
+                integration: {
+                    type: "custom_progress_sitefinity_connector_for_frontify",
```
Comment (FerdiNebi) [inline src/library-extender/dam/custom-dam-provider.ts:40]: same

File: src/library-extender/dam/custom-dam-provider.ts [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+        const apiKey = this.getPropertyValue("ApiKey");
+        if (!apiKey) {
+            return false;
+        }
+
+        return true;
+    }
+
+    private getDamAsset(asset: any): DamAsset {
```
Comment (FerdiNebi) [inline src/library-extender/dam/custom-dam-provider.ts:103]: The implementation is not up to date with the actual implementation for Cloudinary. If we want people to be able to use it as a reference, perhaps we need other changes as well.

**Review by FerdiNebi — APPROVED** [review]

File: src/library-extender/dam/custom-dam-provider.ts (hunk 4 above)
Comment (P1ndl3sk1n) [reply src/library-extender/dam/custom-dam-provider.ts:108]: code is now updated to respect transformations and add sf_cl_ query params in the stored URL

**Pull request closed without merging**